crypto/src/merkle/mod.rs

```diff
@@ -0,0 +1,5 @@
+pub mod constants;
+pub mod hash;
+pub mod node;
+pub mod proof;
+pub mod tree;
```

Comment:
I think it probably makes sense for these modules to be private, and have their contents re-exported out of the `merkle` module (e.g., `merkle::Tree`, `merkle::Proof`, etc) as needed.